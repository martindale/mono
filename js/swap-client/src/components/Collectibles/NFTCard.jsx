import { Grid, Stack } from '@mui/material'
import React from 'react'
import styles from '../../styles/collectibles/index.module.css'

export const NFTCard = ({ card, handleClick }) => {
  return (
    <Grid item xs={3} container direction='column' onClick={e => handleClick(card)}>
      <Stack className={`${styles['nft-card']} nft-card`}>
        <Grid item>
          <img src={card.img_url} width='100%' />
        </Grid>
        <Grid item className='flex-center flex-middle'>
          <h4 className={styles['ordinal-btn']}>
            <img className={styles['btc-logo']} width='20px' src='https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/bitcoin/info/logo.png?raw=true' />Ordinal
          </h4>
        </Grid>
        <Grid item textAlign='center'>
          <h4>{card.info.inscription.slice(0, 7)}</h4>
        </Grid>
      </Stack>
    </Grid>
  )
}
