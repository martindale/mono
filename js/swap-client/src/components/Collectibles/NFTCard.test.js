import { NFTCard } from './NFTCard'
import { create, act } from 'react-test-renderer'

describe('AccessOptionComponent', () => {
  it('should work', () => {
    let tree
    const card = {
      img_url: '123',
      title: 'card'
    }
    const handleClick = jest.fn()

    act(() => {
      tree = create(
        <NFTCard
          card={card}
          handleClick={handleClick}
        />
      )
    })

    expect(tree).toMatchSnapshot()
  })
  afterAll(() => jest.resetModules())
})
