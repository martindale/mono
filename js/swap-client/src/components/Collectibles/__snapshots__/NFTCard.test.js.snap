// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessOptionComponent should work 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3 css-1y066f1-MuiGrid-root"
  onClick={[Function]}
>
  <div
    className="MuiStack-root css-nen11g-MuiStack-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <img
        src="123"
        width="100%"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item flex-center flex-middle css-13i4rnv-MuiGrid-root"
    >
      <h4>
        <img
          src="https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/bitcoin/info/logo.png?raw=true"
          width="20px"
        />
        Ordinal
      </h4>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-10omzie-MuiGrid-root"
    >
      <h4>
        card
      </h4>
    </div>
  </div>
</div>
`;
